In [1]:
import pandas as pd

## loading data

In [2]:
state_minwage_df = pd.read_excel("state_min_wage_rates.xlsx")
state_minwage_df.head()

State or other  2014  2015  2016  2017  2018   2019   2020   2021   2022  \
0    jurisdiction   NaN   NaN   NaN   NaN   NaN    NaN    NaN    NaN    NaN   
1  Federal (FLSA)  7.25  7.25  7.25  7.25  7.25   7.25   7.25   7.25   7.25   
2         Alabama   ...   ...   ...   ...   ...    ...    ...    ...    ...   
3          Alaska  7.75  8.75  9.75   9.8  9.84  10.19  10.19  10.34  10.34   
4         Arizona   7.9  8.05  8.05    10  10.5     12     12  12.15   12.8   

    2023   2024  
0    NaN    NaN  
1   7.25   7.25  
2    ...    ...  
3  10.85  11.73  
4  13.85  14.35

In [3]:
# dropping rows that wont be used
state_minwage_df.drop(0, inplace=True)

In [4]:
state_minwage_df.head(10)

State or other     2014     2015     2016     2017     2018     2019  \
1   Federal (FLSA)     7.25     7.25     7.25     7.25     7.25     7.25   
2          Alabama      ...      ...      ...      ...      ...      ...   
3           Alaska     7.75     8.75     9.75      9.8     9.84    10.19   
4          Arizona      7.9     8.05     8.05       10     10.5       12   
5         Arkansas  6.25[c]  7.50[c]  8.00[c]  8.50[c]  8.50[c]  9.25[c]   
6       California        9        9       10       10       11       12   
7         Colorado        8     8.23     8.31      9.3     10.2       12   
8      Connecticut      8.7     9.15      9.6     10.1     10.1       11   
9         Delaware     7.75     8.25     8.25     8.25     8.25     9.25   
10         Florida     7.93     8.05     8.05      8.1     8.25     8.56   

        2020   2021   2022   2023   2024  
1       7.25   7.25   7.25   7.25   7.25  
2        ...    ...    ...    ...    ...  
3      10.19  10.34  10.34  10.85  11.73  
4         12  12.15   12.8  13.85  14.35  
5   10.00[c]     11     11     11     11  
6         12     13     14   15.5     16  
7      12.02  12.32  12.56  13.65  14.42  
8         12     13     14     15  15.69  
9       9.25   9.25   10.5  11.75  13.25  
10      8.56     10     10     12     12

In [5]:
state_minwage_df.columns

Index(['State or other',             2014,             2015,             2016,
                   2017,             2018,             2019,             2020,
                   2021,             2022,             2023,             2024],
      dtype='object')

### control state selection

In [6]:
state_minwage_df[2024] = pd.to_numeric(state_minwage_df[2024], errors="coerce")
# some values are .., so we coerce
state_minwage_df[state_minwage_df[2024] < 8]

State or other  2014  2015  2016  2017  2018  2019  2020  2021  2022  \
1   Federal (FLSA)  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25   
13           Idaho  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25   
16            Iowa  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25   
17          Kansas  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25   
18        Kentucky  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25   
30   New Hampshire  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25   
34  North Carolina  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25   
35    North Dakota  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25   
39    Pennsylvania  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25   
44           Texas  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25   
45            Utah  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25   
50       Wisconsin  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25  7.25   
51         Wyoming  5.15  5.15  5.15  5.15  5.15  5.15  5.15  5.15  5.15   

    2023  2024  
1   7.25  7.25  
13  7.25  7.25  
16  7.25  7.25  
17  7.25  7.25  
18  7.25  7.25  
30  7.25  7.25  
34  7.25  7.25  
35  7.25  7.25  
39  7.25  7.25  
44  7.25  7.25  
45  7.25  7.25  
50  7.25  7.25  
51  5.15  5.15

> So I think these will be our control states, since they havent made any changes

### treatment state selection

In [7]:
state_minwage_df[state_minwage_df[2024] > 12]

State or other     2014     2015     2016      2017      2018  \
4                Arizona      7.9     8.05     8.05        10      10.5   
6             California        9        9       10        10        11   
7               Colorado        8     8.23     8.31       9.3      10.2   
8            Connecticut      8.7     9.15      9.6      10.1      10.1   
9               Delaware     7.75     8.25     8.25      8.25      8.25   
12                Hawaii     7.25     7.75      8.5      9.25      10.1   
20                 Maine      7.5      7.5      7.5         9        10   
21              Maryland     7.25     8.25     8.75      8.75      10.1   
22         Massachusetts        8        9       10        11        11   
26              Missouri      7.5     7.65     7.65       7.7      7.85   
31            New Jersey     8.25     8.38     8.38      8.44       8.6   
33              New York        8     8.75        9       9.7      10.4   
38                Oregon      9.1     9.25     9.75      9.75     10.75   
40          Rhode Island        8        9      9.6       9.6      10.1   
46               Vermont  8.73(e)  9.15(e)  9.60(e)  10.00(e)  10.50(e)   
48            Washington     9.32     9.47     9.47        11      11.5   
52  District of Columbia      9.5     10.5     11.5      11.5     13.25   

        2019      2020   2021   2022   2023   2024  
4         12        12  12.15   12.8  13.85  14.35  
6         12        12     13     14   15.5  16.00  
7         12     12.02  12.32  12.56  13.65  14.42  
8         11        12     13     14     15  15.69  
9       9.25      9.25   9.25   10.5  11.75  13.25  
12      10.1      10.1   10.1     12     12  14.00  
20        12        12  12.15  12.75   13.8  14.15  
21        11        11  11.75   12.5  13.25  15.00  
22     12.75     12.75   13.5  14.25     15  15.00  
26      9.45      9.45   10.3  11.15     12  12.30  
31        11        11     12     13  14.13  15.13  
33      11.8      11.8   12.5   13.2   14.2  15.00  
38     11.25        12  12.75   13.5   14.2  14.70  
40      10.5      11.5   11.5  12.25     13  14.00  
46  10.96(e)  10.96(e)  11.75  12.55  13.18  13.67  
48      13.5      13.5  13.69  14.49  15.74  16.28  
52        14        15   15.2   16.1     17  17.50

> maybe it would be best to observe the states who also start at around $7.25

> Possible Treatment States: Arizona, Delaware, Hawaii,  Maine, Maryland, New Jersey

In [8]:
state_minwage_df[state_minwage_df["State or other"] == "Florida"]

State or other  2014  2015  2016 2017  2018  2019  2020 2021 2022 2023  \
10        Florida  7.93  8.05  8.05  8.1  8.25  8.56  8.56   10   10   12   

    2024  
10  12.0

# State gdp data loading

In [9]:
# CONTROLS
kansas = pd.read_csv("./state_summarydata/control/kansas.csv")
kentucky = pd.read_csv("./state_summarydata/control/kentucky.csv")
newhampshire = pd.read_csv("./state_summarydata/control/newhampshire.csv")
northcarolina = pd.read_csv("./state_summarydata/control/northcarolina.csv")
pennsylvania = pd.read_csv("./state_summarydata/control/pennsylvania.csv")
texas = pd.read_csv("./state_summarydata/control/texas.csv")
utah = pd.read_csv("./state_summarydata/control/utah.csv")

control_df = pd.concat(
    [kansas, kentucky, newhampshire, northcarolina, pennsylvania, texas, utah],
    ignore_index=True,
)

control_df.tail()

GeoFIPS GeoName  Region  TableName  LineCode IndustryClassification  \
100   "49000"    Utah       7  SASUMMARY        11                    ...   
101   "49000"    Utah       7  SASUMMARY        12                    ...   
102   "49000"    Utah       7  SASUMMARY        13                    ...   
103   "49000"    Utah       7  SASUMMARY        14                    ...   
104   "49000"    Utah       7  SASUMMARY        15                    ...   

                                           Description            Unit  \
100           Per capita disposable personal income 7/         Dollars   
101   Per capita personal consumption expenditures ...         Dollars   
102                  Regional price parities (RPPs) 9/           Index   
103               Implicit regional price deflator 10/           Index   
104                 Total employment (number of jobs)   Number of jobs   

        1998     1999  ...         2015         2016         2017  \
100    19421    20079  ...    35376.000    36513.000    38189.000   
101    18257    18974  ...    31656.000    33009.000    34413.000   
102     (NA)     (NA)  ...       97.656       97.082       98.708   
103     (NA)     (NA)  ...       95.018       95.416       98.708   
104  1309652  1338175  ...  1865934.000  1933131.000  1995267.000   

            2018         2019         2020         2021         2022  \
100    40390.000    42594.000    45667.000    50452.000    51458.000   
101    36272.000    37749.000    38180.000    43895.000    48559.000   
102       95.977       97.259       95.262       94.573       94.572   
103       97.942      100.671       99.683      103.058      109.809   
104  2069738.000  2102960.000  2135682.000  2249697.000  2342012.000   

            2023   2024  
100    55624.000  57962  
101    51027.000   (NA)  
102       95.002   (NA)  
103      114.470   (NA)  
104  2416306.000   (NA)  

[5 rows x 35 columns]

In [15]:
# TREATMENT
arizona = pd.read_csv("./state_summarydata/treatment/arizona.csv")
delaware = pd.read_csv("./state_summarydata/treatment/delaware.csv")
hawaii = pd.read_csv("./state_summarydata/treatment/hawaii.csv")
maine = pd.read_csv("./state_summarydata/treatment/maine.csv")
maryland = pd.read_csv("./state_summarydata/treatment/maryland.csv")
newjersey = pd.read_csv("./state_summarydata/treatment/newjersy.csv")

treated_df = pd.concat(
    [arizona, delaware, hawaii, maine, maryland, newjersey],
    ignore_index=True,
)

treated_df.head()

GeoFIPS  GeoName  Region  TableName  LineCode IndustryClassification  \
0   "04000"  Arizona       6  SASUMMARY         1                    ...   
1   "04000"  Arizona       6  SASUMMARY         2                    ...   
2   "04000"  Arizona       6  SASUMMARY         3                    ...   
3   "04000"  Arizona       6  SASUMMARY         4                    ...   
4   "04000"  Arizona       6  SASUMMARY         5                    ...   

                                         Description  \
0     Real GDP (millions of chained 2017 dollars) 1/   
1   Real personal income (millions of constant (2...   
2   Real PCE (millions of constant (2017) dollars...   
3                      Gross domestic product (GDP)    
4                                   Personal income    

                                Unit      1998      1999  ...      2015  \
0   Millions of chained 2017 dollars  197492.1  214293.9  ...  308582.8   
1  Millions of constant 2017 dollars      (NA)      (NA)  ...  281347.5   
2  Millions of constant 2017 dollars      (NA)      (NA)  ...  246519.2   
3        Millions of current dollars  144393.2  156442.1  ...  298942.2   
4        Millions of current dollars  117835.0  124922.3  ...  267474.1   

       2016      2017      2018      2019      2020      2021      2022  \
0  319008.4  333099.0  346398.3  359576.7  365027.7  395035.9  410228.4   
1  289273.8  304434.3  318218.3  333868.0  361266.1  394399.4  374592.8   
2  251483.4  264659.3  275644.2  283289.2  281220.9  318965.7  321902.1   
3  313787.1  333099.0  353671.0  375376.7  388956.7  436778.6  484052.2   
4  279696.7  297261.5  315459.0  337257.2  374007.2  414355.4  434338.9   

       2023      2024  
0  422399.6  433763.2  
1  381769.1      (NA)  
2  328923.0      (NA)  
3  522767.2  552167.0  
4  464774.9  488730.4  

[5 rows x 35 columns]

In [10]:
# dropping unnecessary columns
control_df = control_df.drop(["GeoFIPS", "Region", "TableName", "LineCode"], axis=1)
treated_df = treated_df.drop(["GeoFIPS", "Region", "TableName", "LineCode"], axis=1)

GeoName IndustryClassification  \
0  Kansas                    ...   
1  Kansas                    ...   
2  Kansas                    ...   
3  Kansas                    ...   
4  Kansas                    ...   

                                         Description  \
0     Real GDP (millions of chained 2017 dollars) 1/   
1   Real personal income (millions of constant (2...   
2   Real PCE (millions of constant (2017) dollars...   
3                      Gross domestic product (GDP)    
4                                   Personal income    

                                Unit      1998      1999      2000      2001  \
0   Millions of chained 2017 dollars  112625.7  115486.4  118840.9  119474.1   
1  Millions of constant 2017 dollars      (NA)      (NA)      (NA)      (NA)   
2  Millions of constant 2017 dollars      (NA)      (NA)      (NA)      (NA)   
3        Millions of current dollars   78242.3   81609.7   85990.5   88869.7   
4        Millions of current dollars   69711.2   71968.6   75870.8   78490.8   

       2002      2003  ...      2015      2016      2017      2018      2019  \
0  121050.5  123486.7  ...  158602.6  164643.7  166273.6  169796.9  169562.3   
1      (NA)      (NA)  ...  150962.5  151347.9  154694.8  156470.6  159653.1   
2      (NA)      (NA)  ...  112680.2  115222.0  117367.2  117900.6  119196.3   
3   91359.0   95236.3  ...  155699.0  162144.5  166273.6  173373.0  176068.8   
4   79285.4   81768.2  ...  136189.1  136753.6  141311.8  147407.4  153011.9   

       2020      2021      2022      2023      2024  
0  168056.8  172782.4  177470.1  183795.6  185720.9  
1  168145.8  174737.3  175964.6  179467.7      (NA)  
2  117488.2  125880.6  131627.4  133865.8      (NA)  
3  177209.6  191738.1  212611.6  228232.4  234673.2  
4  161892.4  173399.5  183033.7  194413.5  202115.4  

[5 rows x 31 columns]

In [17]:
control_df.columns

Index(['GeoName', 'IndustryClassification', 'Description', 'Unit', '1998',
       '1999', '2000', '2001', '2002', '2003', '2004', '2005', '2006', '2007',
       '2008', '2009', '2010', '2011', '2012', '2013', '2014', '2015', '2016',
       '2017', '2018', '2019', '2020', '2021', '2022', '2023', '2024'],
      dtype='object')

In [18]:
# filter to only include same years as minimum wage dataset
control_df = control_df.drop(
    [
        "1998",
        "1999",
        "2000",
        "2001",
        "2002",
        "2003",
        "2004",
        "2005",
        "2006",
        "2007",
        "2008",
        "2009",
        "2010",
        "2011",
        "2012",
        "2013",
    ],
    axis=1,
)

treated_df = treated_df.drop(
    [
        "1998",
        "1999",
        "2000",
        "2001",
        "2002",
        "2003",
        "2004",
        "2005",
        "2006",
        "2007",
        "2008",
        "2009",
        "2010",
        "2011",
        "2012",
        "2013",
    ],
    axis=1,
)

control_df.head()

GeoName IndustryClassification  \
0  Kansas                    ...   
1  Kansas                    ...   
2  Kansas                    ...   
3  Kansas                    ...   
4  Kansas                    ...   

                                         Description  \
0     Real GDP (millions of chained 2017 dollars) 1/   
1   Real personal income (millions of constant (2...   
2   Real PCE (millions of constant (2017) dollars...   
3                      Gross domestic product (GDP)    
4                                   Personal income    

                                Unit      2008      2009      2010      2011  \
0   Millions of chained 2017 dollars  147265.0  141902.5  143870.1  147836.8   
1  Millions of constant 2017 dollars  144002.4  136817.7  135439.8  143276.0   
2  Millions of constant 2017 dollars  110663.1  106339.3  105773.7  107580.8   
3        Millions of current dollars  129223.2  126049.7  129235.7  136747.1   
4        Millions of current dollars  115298.3  111558.9  113843.0  123198.9   

       2012      2013  ...      2015      2016      2017      2018      2019  \
0  150077.1  150702.9  ...  158602.6  164643.7  166273.6  169796.9  169562.3   
1  147397.5  149555.2  ...  150962.5  151347.9  154694.8  156470.6  159653.1   
2  107494.9  108661.1  ...  112680.2  115222.0  117367.2  117900.6  119196.3   
3  141812.1  144999.3  ...  155699.0  162144.5  166273.6  173373.0  176068.8   
4  129849.7  132519.0  ...  136189.1  136753.6  141311.8  147407.4  153011.9   

       2020      2021      2022      2023      2024  
0  168056.8  172782.4  177470.1  183795.6  185720.9  
1  168145.8  174737.3  175964.6  179467.7      (NA)  
2  117488.2  125880.6  131627.4  133865.8      (NA)  
3  177209.6  191738.1  212611.6  228232.4  234673.2  
4  161892.4  173399.5  183033.7  194413.5  202115.4  

[5 rows x 21 columns]